compiler/src/dmd/lexer.d

```diff
@@ -1972,6 +1972,13 @@ class Lexer
         case 'd':
             t.postfix = *p;
             p++;
+            // diagnose e.g. `@r"_"dtype var;`
+            if (!Ccompile && (isidchar(*p) || *p & 0x80))
```

Comment:
Checking for `*p & 0x80` will also produce a warning for unicode line and paragraph separators. I don't know if anybody uses them, but they are currently allowed: https://dlang.org/spec/lex.html#end_of_line
```
  Foo!q{foo}c   c;
  Foo!q{foo}b   b;
```